From d06da9403d144a6e3f612b5f98fd781cee3889b7 Mon Sep 17 00:00:00 2001
From: Finley Xiao <finley.xiao@rock-chips.com>
Date: Wed, 1 Apr 2020 11:29:03 +0800
Subject: [PATCH] clk: rockchip: rv1126: Add pll clocks to critical

As the all parent of free mux should be enabled when set rate, and the
clk framework does not support to enable all parent of mux with more then
two parent.

Change-Id: Iee6741f2829d47ddaba05438774051691318c7b9
Signed-off-by: Finley Xiao <finley.xiao@rock-chips.com>
---
 drivers/clk/rockchip/clk-rv1126.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/clk/rockchip/clk-rv1126.c b/drivers/clk/rockchip/clk-rv1126.c
index 35a465971d8d..0c31f1b36b49 100644
--- a/drivers/clk/rockchip/clk-rv1126.c
+++ b/drivers/clk/rockchip/clk-rv1126.c
@@ -1281,6 +1281,9 @@ static struct rockchip_clk_branch rv1126_clk_branches[] __initdata = {
 };
 
 static const char *const rv1126_cru_critical_clocks[] __initconst = {
+	"gpll",
+	"cpll",
+	"hpll",
 	"armclk",
 	"pclk_dbg",
 	"pclk_pdpmu",
-- 
2.35.3

